### PR TITLE
init: add --from-code flag

### LINKED ISSUE #3361: Need a flag for `azd init` that will use the current directory instead of selecting a template
The new `azd init` with the functionality to use the source in the current directory does not seem to have a flag that can be passed to cause it to take that behavior. I don't think `--no-prompt` will work as that would suppress other prompts that the CLI needs to make.

### PR DESCRIPTION
**Pull request by weikanglim**
Add a flag to `init` for initializing from existing code.

Fixes #3361

### CODE REVIEW

File: cli/azd/cmd/init.go [hunk 1]
```diff
@@ -78,6 +78,13 @@ func (i *initFlags) Bind(local *pflag.FlagSet, global *internal.GlobalCommandOpt
 		"",
 		"Name or ID of an Azure subscription to use for the new environment",
 	)
+	local.BoolVarP(
+		&i.fromCode,
+		"from-code",
+		"",
+		false,
+		"Initializes a new application from your existing code.",
+	)
```
Comment (vhvb1989) [inline cli/azd/cmd/init.go:87]: Should we use the pattern:

```go
f := local.VarPF(...)
// ensure the flag behaves as a common boolean flag which is set to true when used without any other arg
f.NoOptDefVal = "true"
```

That's what we do for a few bool args around there, like `use-device-code`

File: cli/azd/cmd/init.go [hunk 2]
```diff
@@ -53,6 +53,7 @@ type initFlags struct {
 	subscription   string
 	location       string
 	global         *internal.GlobalCommandOptions
+	fromCode       bool
```
[the remark this answers is not in the document]
Comment (weikanglim) [reply cli/azd/cmd/init.go:56]: Thanks for the mention! IIRC, the team had some questions about the scenario, and we weren't in full consensus on supporting it yet. Will happily submit a separate change when we decide. I've started the discussion on the issue.

File: cli/azd/cmd/init.go (hunk 1 above)
Comment (ellismg) [reply cli/azd/cmd/init.go:87]: I'm not 100% sure what that pattern buys us?   Is that just about configuring the default when it is not on the command line? If we wanted to default the value to true, I think we would set the `false` to `true` on L85.

FWIW, I prefer what @weikanglim has here if they are semantically the same.

**Review by ellismg — APPROVED** [review]

File: cli/azd/cmd/init.go (hunk 1 above)
Comment (vhvb1989) [reply cli/azd/cmd/init.go:87]: >  I think we would set the false to true on L85.

That would mean that if you don't set the flag, at all, you will get `true`, which is not what we want, @ellismg 

IIRC, what we get from that pattern is the ability to know if the flag is false b/c it was not manually added, or if the flag was added like `azd init --from-code=false`.

Comment (weikanglim) [reply cli/azd/cmd/init.go:87]: The `boolPtr` pattern is documented as follows:

> // boolPtr implements a pflag.Value and allows us to distinguish between a flag value being explicitly set to
// bool vs not being present.

So, we'd adopt the pattern if we need to distinguish between `--from-code 0` versus unset. In `auth login`, we want:

```bash
$ azd auth login # unset, but can be autodetermined by azd
$ azd auth login --use-device-code=false # explicitly set, azd will not modify behavior
```

 but we don't need that here, is what I'm thinking.

Comment (vhvb1989) [reply cli/azd/cmd/init.go:87]: > but we don't need that here

Shouldn't we collect telemetry usage for all invocations?  Maybe we will wonder in the future, how many folks are passing bool flags explicitly v/s letting azd to set defaults. :)

**Review by vhvb1989 — APPROVED** [review]
Cool!

**Pull request merged**